mod-source-record-manager-server/src/main/java/org/folio/dao/MappingRulesSnapshotDaoImpl.java

```diff
@@ -0,0 +1,80 @@
+package org.folio.dao;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import io.vertx.sqlclient.Tuple;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dao.util.PostgresClientFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+import static java.lang.String.format;
+import static org.folio.rest.persist.PostgresClient.convertToPsqlStandard;
+
+@Repository
+public class MappingRulesSnapshotDaoImpl implements MappingRulesSnapshotDao {
+
+  @Autowired
+  private PostgresClientFactory pgClientFactory;
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  private static final String TABLE_NAME = "mapping_rules_snapshots";
+  private static final String SELECT_QUERY = "SELECT rules FROM %s.%s WHERE job_execution_id = $1";
+  private static final String INSERT_SQL = "INSERT INTO %s.%s (job_execution_id, rules, saved_timestamp) VALUES ($1, $2, $3)";
+  private static final String DELETE_BY_JOB_EXECUTION_ID_QUERY = "DELETE FROM %s.%s WHERE job_execution_id = $1";
+
+  @Override
+  public Future<Optional<JsonObject>> getByJobExecutionId(String jobExecutionId, String tenantId) {
+    Promise<RowSet<Row>> promise = Promise.promise();
+    String query = format(SELECT_QUERY, convertToPsqlStandard(tenantId), TABLE_NAME);
+    Tuple queryParams = Tuple.of(UUID.fromString(jobExecutionId));
+    pgClientFactory.createInstance(tenantId).select(query, queryParams, promise);
+    return promise.future().map(resultSet -> {
+      if (resultSet.rowCount() == 0) {
+        return Optional.empty();
+      } else {
+        JsonObject rules = new JsonObject(resultSet.iterator().next().getValue("rules").toString());
+        return Optional.of(rules);
+      }
+    });
+  }
+
+  @Override
+  public Future<String> save(JsonObject rules, String jobExecutionId, String tenantId) {
+    Promise<RowSet<Row>> promise = Promise.promise();
+    try {
+      String query = format(INSERT_SQL, convertToPsqlStandard(tenantId), TABLE_NAME);
+      Tuple queryParams = Tuple.of(
+        UUID.fromString(jobExecutionId),
+        rules,
+        LocalDateTime.now()
```

Comment:
I think it would be better to add timezones, to avoid potential bugs as we already caught for another mechanism. To add timezone we need to make a specific request to another service: https://github.com/folio-org/mod-source-record-manager/blob/483bb7ba358c6daee8d87c6600d94d8f1c902ae9/mod-source-record-manager-server/src/main/java/org/folio/services/mappers/processor/MappingParametersProvider.java#L696

Reply:
we don't care that much for a time when the metadata was saved, it might be useful when we decide to clear the DB of some old irrelevant data, and timezone won't make any difference